app/js/showControllers.js

```diff
@@ -46,6 +46,13 @@ angular.module('modulusOne.showControllers', [
       })
     }
 
+    $scope.cancelEdit = function(toggleEditWhenDone) {
+      $state.transitionTo($state.current, $stateParams, {
+        reload: true, inherit: false, notify: true
+      });
+
```

Comment:
$state.reload worked well unless you didn't "tick" the changes. $state.reload is an alias for $state.transitionTo (https://angular-ui.github.io/ui-router/site/#/api/ui.router.state.$state#methods_reload). $state.transitionTo function has the desired effect.